ios-cantact-manager/ios-cantact-manager/ContactManageSystem.swift

```diff
@@ -9,24 +9,81 @@ import Foundation
 
 struct ContactManageSystem {
     let inputManager = InputManager()
-    var contactList = [String: Profile]()
+    var profileData = Set<Profile>()
+    var isFinished = false
+    
+    enum Menu: String {
+        case addProfile = "1"
+        case listUpProfile = "2"
+        case searchProfile = "3"
+        case stop = "x"
+    }
+    
+    func deliverMenuInputValue() throws -> String {
+        OutputManager.printMessage(.inputMenu)
+        let menuInput = try inputManager.menuInput()
+        
+        return menuInput
+    }
+    
+    mutating func pipeInMenu(_ input: String) {
+            guard let input = Menu(rawValue: input) else {
+                OutputManager.printMessage(.invalidMenu)
+                return
+            }
+            switch input {
+            case .addProfile:
+                addProfile()
+            case .listUpProfile:
+                listUpProfile()
+            case .searchProfile:
+                searchProfile()
+            case .stop:
+                stop()
+            }
+        }
     
     mutating func addProfile() {
         do {
+            OutputManager.printMessage(.inputInfo)
             let inputArray = try inputManager.parseUserInput()
             let (name, age ,tel) = (inputArray[0], inputArray[1], inputArray[2])
             try inputManager.checkUserInput(name, age, tel)
             let profile = Profile(name: name, age: age, tel: tel)
-            contactList.updateValue(profile, forKey: name)
-            print("입력한 정보는 \(age)세 \(name)(\(tel))입니다.")
+            profileData.insert(profile)
+            OutputManager.printProfile(profile)
         } catch InputError.invalidInput {
-            print("입력한 정보가 잘못되었습니다. 입력 형식을 확인해주세요.")
+            OutputManager.printMessage(.invalidInput)
         } catch InputError.invalidAge {
-            print("입력한 나이정보가 잘못되었습니다. 입력 형식을 확인해주세요.")
+            OutputManager.printMessage(.invalidAge)
         } catch InputError.invalidTel {
-            print("입력한 연락처정보가 잘못되었습니다. 입력 형식을 확인해주세요.")
+            OutputManager.printMessage(.invalidTel)
         } catch {
-            print("입력한 정보가 잘못되었습니다. 입력 형식을 확인해주세요.")
+            OutputManager.printMessage(.invalidInput)
         }
     }
+    
+    func listUpProfile() {
+        OutputManager.printProfileList(profileData)
+    }
+    
+    func searchProfile() {
+        do {
+            OutputManager.printMessage(.inputProfileName)
+            let targetInput = try inputManager.targetInput()
+            let filteredProfileData = profileData.filter { $0.name == targetInput }
+            guard filteredProfileData.isEmpty else {
+                OutputManager.printProfileList(filteredProfileData)
```

Comment:
파라미터와 함수명이 중복되지 않게 다음과 같이 표기하는게, Swift 네이밍 가이드에 준수하는거에요! 😊
```suggestion
                OutputManager.print(profiles: profiles)